simple("${exchange.getMessage().getBody()}") causes a MethodNotFoundException in native mode

# Bug description
Using `simple("${exchange.getMessage().getBody()}"` throws an exception at runtime when compiling natively:

```
Caused by: org.apache.camel.component.bean.MethodNotFoundException: Method with name: getMessage() not found on bean: Exchange[ED6D02F79FBB419-0000000000000000] of type: org.apache.camel.support.DefaultExchange on the exchange: Exchange[ED6D02F79FBB419-0000000000000000]
...
```

# Affected versions:
Quarkus,  at least from `2.13.x.Final` up to the current version (`3.0.1.Final`). 

# Reproducer:
- clone [this repository (`github.com`)](https://github.com/turing85/camel-simple-language-bug):
    ```bash
    git clone https://github.com/turing85/camel-simple-language-bug.git
    cd camel-simple-language-bug
    ```
- Build the application:
    ```bash
    ./mvnw clean verify
    ```
- Observe that the tests pass:
    ```bash
    ...
    [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
    ...
    ```
- Build the application natively:
    ```bash
    ./mvn --define native clean verify
    ```
- Observe that the tests fail with the above mentioned error in the application log, and an unexpected status code on the tests, reporting a status of `500` instead of the expected `200`:
    ```bash
    ...
    Caused by: org.apache.camel.component.bean.MethodNotFoundException: Method with name: getMessage() not found on bean: Exchange[FD7D7D11C0B4B98-0000000000000000] of type: org.apache.camel.support.DefaultExchange on the exchange: Exchange[FD7D7D11C0B4B98-0000000000000000]
    ...
    Expected status code <200> but was <500>.
    ...
    ```

# Workaround
Add the following to `application.yml`

```yaml
quarkus:
  camel:
    native:
      reflection:
        include-patterns:
          - org.apache.camel.support.AbstractExchange
          - org.apache.camel.support.MessageSupport
```

# Possible solution
The corresponding extension should register those classes for reflection by default. This, however, begs the question: what else should be registered? We should analyze all relevant classes used by the `simple(...)` language and ensure that they all work as expected.

--- a/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java
+++ b/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java
@@ -139,6 +139,11 @@ public class CamelNativeImageProcessor {
                 .map(ai -> ai.target().asMethod())
                 .forEach(mi -> reflectiveMethod.produce(new ReflectiveMethodBuildItem(mi)));
 
+        reflectiveClass.produce(
+                ReflectiveClassBuildItem.builder(
+                        "org.apache.camel.support.AbstractExchange",
+                        org.apache.camel.support.MessageSupport.class.getName())
+                        .methods().build());
     }
 
     @BuildStep

--- a/integration-test-groups/foundation/core-languages/src/main/java/org/apache/camel/quarkus/core/languages/it/SimpleRoutes.java
+++ b/integration-test-groups/foundation/core-languages/src/main/java/org/apache/camel/quarkus/core/languages/it/SimpleRoutes.java
@@ -36,6 +36,9 @@ public class SimpleRoutes extends RouteBuilder {
         from("direct:languageSimple")
                 .transform().language("simple", "Hello ${body} from language().simple()");
 
+        from("direct:simpleExchangeMethods")
+                .transform().simple("exchange.getMessage().getBody() evals to ${exchange.getMessage().getBody()}");
+
     }
 
 }

--- a/integration-test-groups/foundation/core-languages/src/test/java/org/apache/camel/quarkus/core/languages/it/SimpleTest.java
+++ b/integration-test-groups/foundation/core-languages/src/test/java/org/apache/camel/quarkus/core/languages/it/SimpleTest.java
@@ -69,10 +69,10 @@ public class SimpleTest {
     }
 
     @Test
-    public void languageSimple() {
-        given().body("Frankenstein").when().post("/core-languages/route/languageSimple/String").then()
+    public void simpleExchangeMethods() {
+        given().body("Joe").when().post("/core-languages/route/simpleExchangeMethods/String").then()
                 .statusCode(200)
-                .body(is("Hello Frankenstein from language().simple()"));
+                .body(is("exchange.getMessage().getBody() evals to Joe"));
     }
 
 }

